src/game/Game.java

```diff
@@ -107,98 +103,164 @@ public int getPlayerTurn() {
+							&& this.getPieceAt(coor).getOwner() == player
+							&& this.getPieceAt(coor).isStrongerThan(this.getPieceAt(pieceToMove))) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+	
+	private boolean isNextToFriendlyPiece(Coordinate pieceToMove, Owner player) {
+		for (int i = -1; i < 2; i++) {
+			for (int j = -1; j < 2; j++) {
+				Coordinate coor = new Coordinate(i + pieceToMove.getX(), j + pieceToMove.getY());
+				if (this.getPieceAt(coor) != null) {
+					if (coor.isValid() && !coor.equals(pieceToMove)
+							&& this.getPieceAt(coor).getOwner() == player) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+	
 	/**
 	 * 
-	 * @param moveToMake
-	 * @param row
-	 * @param column
-	 * @return returns true if the move made successfully, otherwise returns false
+	 * @param ownerPiece
+	 *            Coordinate of the owner's piece
+	 * @param opponentPiece
+	 *            Coordinate of the opponent's piece
+	 * @param destination
+	 *            Coordinate of the position that either the opponent's piece
+	 *            will be pushed into or the position the owner's piece will be
+	 *            moved into.
+	 * @return Returns true when a push or pull with the given 3 Coordinate
+	 *         objects would result in a valid move.
 	 */
 
-	private boolean makeMove(MoveCommand moveToMake, int row, int column) {
-		if (moveToMake.isValidMove(row, column)) {
-			this.currentBoard = moveToMake.execute(row, column);
-			this.moves.add(moveToMake);
-			endMove();
-			return true;
+	public boolean pushOrPull(Coordinate ownerPiece, Coordinate opponentPiece, Coordinate destination) {
+		// Do you have enough moves?
+		if (this.numMoves >= 2) {
+			// Is the first piece yours, the second theirs, are they
+			// orthonally adjacent, and is your piece higher precedence than
+			// theirs?
+			if ((this.currentBoard.getPieceAt(ownerPiece)).getOwner() == this.getOwner()
+					&& (this.currentBoard.getPieceAt(opponentPiece).getOwner() != this.getOwner())
+					&& (ownerPiece.isOrthogonallyAdjacentTo(opponentPiece))
+					&& (this.checkStrongerAdjacent(ownerPiece, opponentPiece))) {
+				// Is the destination next to your piece?
+				if (ownerPiece.isOrthogonallyAdjacentTo(destination)) {
+					RegularMove yourPiece = new RegularMove(this.currentBoard, ownerPiece, destination);
+					RegularMove theirPiece = new RegularMove(this.currentBoard, opponentPiece, ownerPiece);
+					pushOrPullMove(yourPiece, theirPiece);
+					return true;
+					// Or is it next to their piece?
+				} else if (opponentPiece.isOrthogonallyAdjacentTo(destination)) {
+					RegularMove theirPiece = new RegularMove(this.currentBoard, opponentPiece, destination);
+					RegularMove yourPiece = new RegularMove(this.currentBoard, ownerPiece, opponentPiece);
+					pushOrPullMove(theirPiece, yourPiece);
+					return true;
+				} else
+					// Neither, so that isn't a legal move
+					return false;
+			}
```

Comment:
Duplicate Code refactor from separate push/pull logic to a singular "PushOrPull" method, as each operation is similar.